ground/src/main/java/com/google/android/ground/persistence/local/LocalValueStore.kt

```diff
@@ -33,13 +39,23 @@ import timber.log.Timber
  * implementation.
  */
 @Singleton
-class LocalValueStore @Inject constructor(private val preferences: SharedPreferences) {
+class LocalValueStore
+@Inject
+constructor(
+  private val preferences: SharedPreferences,
+  @ApplicationScope private val externalScope: CoroutineScope
+) {
   private val mapTypeProcessor: BehaviorProcessor<MapType> =
     BehaviorProcessor.createDefault(mapType)
 
+  private val _offlineImageryEnabled = MutableStateFlow(isOfflineImageryEnabled)
+
   val mapTypeFlowable: Flowable<MapType>
     get() = allowThreadDiskReads { mapTypeProcessor }
 
+  val offlineImageryEnabledFlow: SharedFlow<Boolean> =
```

Comment:
Since this is a state, perhaps it should be modeled as a `StateFlow`? Then `shareIn` won't be needed.

Reply:
Done